File: Matlab_Source_Code - JT/Python - JT/calcSeasonalRate.py
import numpy as np

def calcSeasonalRate(prices, months, daysInMonth):
    Cbuy = np.zeros(8760)
    hCount = 0

    for m in range(12):
        hoursStart = hCount
        hoursEnd = hoursStart + (24 * daysInMonth[m])
        hoursRange = list(range(hoursStart, hoursEnd))
        Cbuy[hoursRange] = prices[months[m]]
        hCount = hoursEnd

    return Cbuy

np.savetxt("Cbuy.csv", calcSeasonalRate([0.08, 0.12], [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0], [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]))

File: Backend Codes/MATLAB final version/calcSeasonalRate.m
function [Cbuy]=calcSeasonalRate(prices, months, daysInMonth)
    Cbuy = zeros(1,8760);
    hCount = 1;

    for m=1:12
        hoursStart = hCount;
        hoursEnd = hoursStart + (24 * daysInMonth(m));
        hoursRange = hoursStart:hoursEnd-1;
        Cbuy(hoursRange) = prices{1}(months(m));
        hCount = hoursEnd ; 

    end
end